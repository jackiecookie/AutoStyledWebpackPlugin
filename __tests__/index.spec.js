const webpack = require("webpack")
const path = require("path");
const fs = require("fs");
const rimraf = require("rimraf");
const { GetWebpackConfig } = require("./config");
const InjectCssWebpackPlugin = require("../lib/index");


const OUTPUT_DIR = "dist";

jest.setTimeout(300000);

describe("test", () => {
  beforeAll(done => {
    rimraf(path.join(__dirname, OUTPUT_DIR), done);
  });


  test("new plugin", () => {
    expect(typeof (new InjectCssWebpackPlugin().apply)).toEqual("function")
  })

  function testPlugin(name, library, done, expectModule, style = "style") {
    const webpackConfig = GetWebpackConfig(name, OUTPUT_DIR, { library, style }, __dirname)
    let outputFile = `${name}`;
    webpack(webpackConfig, (err, state) => {
      // if (err) {
      //   throw err
      // }
      // console.log('state', state.toString())
      let outputFilePath = path.join(__dirname, OUTPUT_DIR, outputFile);
      const outputFileExists = fs.existsSync(outputFilePath);
      expect(outputFileExists).toBe(true);
      const content = fs.readFileSync(outputFilePath).toString();
       expect(content).toMatchSnapshot(name);
      let { exist = [], notExist = [] } = expectModule;
      let modules = state.compilation.modules;
      exist.every(existModule => {
        let index = modules.findIndex(module => {
          return new RegExp(existModule + "$").test(module.request);
        });
        expect(index).toBeGreaterThan(-1);
      });

      notExist.every(notexistModule => {
        let index = modules.findIndex(module => {
          return new RegExp(notexistModule + "$").test(module.request);
        });
        expect(index).toEqual(-1);
      });
      done();
    });
  }
  const library = "./element";
  test("test normal import file", done => {
    testPlugin(
      "normalImportFile.js",
      library,
      done,
      {
        exist: ["button.css", "message.css"]
      },
      function style(rawRequest, name) {
        return `${rawRequest}/style/${name.toLowerCase()}.css`;
      }
    );
  });

  test("test normal import file and get tree shaking", done => {
    testPlugin("normalImportWithTreeeShaking.js", library, done, {
      exist: ["button.css"],
      notExist: ["message.css"]
    });
  });

  test("test dynamic import", done => {
    testPlugin("dynamicImport.js", library, done, {
      exist: ["button.css", "message.css"]
    });
  });

  test("test dynamic import complex", done => {
    testPlugin("dynamicImportComplex.js", library, done, {
      exist: ["button.css", "message.css"]
    });
  });
});
