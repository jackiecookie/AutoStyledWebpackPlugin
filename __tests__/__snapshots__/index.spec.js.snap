// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test test dynamic import complex: dynamicImportComplex.js 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./__tests__/support/dynamicImportComplex.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./__tests__/support/dynamicImportComplex.js\\":
/*!***************************************************!*\\\\
  !*** ./__tests__/support/dynamicImportComplex.js ***!
  \\\\***************************************************/
/*! exports provided: default */
/*! all exports used */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"__webpack_require__.r(__webpack_exports__);\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE__ = __webpack_require__(/*! ./__tests__/support/element/style/button.css */ \\\\\\"./__tests__/support/element/style/button.css\\\\\\");\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE___default = /*#__PURE__*/__webpack_require__.n(_Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE__);\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE__ = __webpack_require__(/*! ./__tests__/support/element/style/message.css */ \\\\\\"./__tests__/support/element/style/message.css\\\\\\");\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE___default = /*#__PURE__*/__webpack_require__.n(_Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE__);\\\\n// let name = './element/button'\\\\n\\\\n\\\\nlet name1 = './element/message';\\\\nfunction a(){\\\\n    let name = './element/button'\\\\n    {\\\\n        let name = './element/button2'\\\\n    }\\\\n    return {\\\\n        components: {\\\\n            'component': () => import(name),\\\\n            'componenta': () => import(name1)\\\\n        }\\\\n    }\\\\n    name = ''\\\\n}\\\\n\\\\n/* harmony default export */ __webpack_exports__[\\\\\\"default\\\\\\"] = (a());\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/dynamicImportComplex.js?\\");

/***/ }),

/***/ \\"./__tests__/support/element/style/button.css\\":
/*!****************************************************!*\\\\
  !*** ./__tests__/support/element/style/button.css ***!
  \\\\****************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

eval(\\"// \\\\\\".button{\\\\\\\\n    color: red;\\\\\\\\n}\\\\\\"\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/style/button.css?\\");

/***/ }),

/***/ \\"./__tests__/support/element/style/message.css\\":
/*!*****************************************************!*\\\\
  !*** ./__tests__/support/element/style/message.css ***!
  \\\\*****************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

eval(\\"// \\\\\\".message{\\\\\\\\n    color: red;\\\\\\\\n}\\\\\\"\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/style/message.css?\\");

/***/ })

/******/ });"
`;

exports[`test test dynamic import: dynamicImport.js 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./__tests__/support/dynamicImport.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./__tests__/support/dynamicImport.js\\":
/*!********************************************!*\\\\
  !*** ./__tests__/support/dynamicImport.js ***!
  \\\\********************************************/
/*! exports provided: default */
/*! all exports used */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"__webpack_require__.r(__webpack_exports__);\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE__ = __webpack_require__(/*! ./__tests__/support/element/style/button.css */ \\\\\\"./__tests__/support/element/style/button.css\\\\\\");\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE___default = /*#__PURE__*/__webpack_require__.n(_Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE__);\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE__ = __webpack_require__(/*! ./__tests__/support/element/style/message.css */ \\\\\\"./__tests__/support/element/style/message.css\\\\\\");\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE___default = /*#__PURE__*/__webpack_require__.n(_Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE__);\\\\nlet name = './element/button'\\\\n\\\\n\\\\n/* harmony default export */ __webpack_exports__[\\\\\\"default\\\\\\"] = ({\\\\n    components: {\\\\n        'component': () => import(name),\\\\n        'componenta': () => import('./element/message')\\\\n    }\\\\n});\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/dynamicImport.js?\\");

/***/ }),

/***/ \\"./__tests__/support/element/style/button.css\\":
/*!****************************************************!*\\\\
  !*** ./__tests__/support/element/style/button.css ***!
  \\\\****************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

eval(\\"// \\\\\\".button{\\\\\\\\n    color: red;\\\\\\\\n}\\\\\\"\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/style/button.css?\\");

/***/ }),

/***/ \\"./__tests__/support/element/style/message.css\\":
/*!*****************************************************!*\\\\
  !*** ./__tests__/support/element/style/message.css ***!
  \\\\*****************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

eval(\\"// \\\\\\".message{\\\\\\\\n    color: red;\\\\\\\\n}\\\\\\"\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/style/message.css?\\");

/***/ })

/******/ });"
`;

exports[`test test normal import file and get tree shaking: normalImportWithTreeeShaking.js 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./__tests__/support/normalImportWithTreeeShaking.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./__tests__/support/element/button.js\\":
/*!*********************************************!*\\\\
  !*** ./__tests__/support/element/button.js ***!
  \\\\*********************************************/
/*! exports provided: default */
/*! exports used: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"\\\\nlet Button = 'Button'\\\\n/* harmony default export */ __webpack_exports__[\\\\\\"a\\\\\\"] = (Button);\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/button.js?\\");

/***/ }),

/***/ \\"./__tests__/support/element/index.js\\":
/*!********************************************!*\\\\
  !*** ./__tests__/support/element/index.js ***!
  \\\\********************************************/
/*! exports provided: Message, Button */
/*! exports used: Button */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"/* harmony import */ var _button__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./button */ \\\\\\"./__tests__/support/element/button.js\\\\\\");\\\\n/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\\\\\"a\\\\\\", function() { return _button__WEBPACK_IMPORTED_MODULE_0__[\\\\\\"a\\\\\\"]; });\\\\n\\\\n/* harmony import */ var _message__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ./message */ \\\\\\"./__tests__/support/element/message.js\\\\\\");\\\\n\\\\n\\\\n\\\\n\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/index.js?\\");

/***/ }),

/***/ \\"./__tests__/support/element/message.js\\":
/*!**********************************************!*\\\\
  !*** ./__tests__/support/element/message.js ***!
  \\\\**********************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"\\\\nlet Message = 'Message'\\\\n/* unused harmony default export */ var _unused_webpack_default_export = (Message);\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/message.js?\\");

/***/ }),

/***/ \\"./__tests__/support/element/style/button.css\\":
/*!****************************************************!*\\\\
  !*** ./__tests__/support/element/style/button.css ***!
  \\\\****************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

eval(\\"// \\\\\\".button{\\\\\\\\n    color: red;\\\\\\\\n}\\\\\\"\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/style/button.css?\\");

/***/ }),

/***/ \\"./__tests__/support/normalImportWithTreeeShaking.js\\":
/*!***********************************************************!*\\\\
  !*** ./__tests__/support/normalImportWithTreeeShaking.js ***!
  \\\\***********************************************************/
/*! no exports provided */
/*! all exports used */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"__webpack_require__.r(__webpack_exports__);\\\\n/* harmony import */ var _element__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./element */ \\\\\\"./__tests__/support/element/index.js\\\\\\");\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE__ = __webpack_require__(/*! ./__tests__/support/element/style/button.css */ \\\\\\"./__tests__/support/element/style/button.css\\\\\\");\\\\n/* harmony import */ var _Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE___default = /*#__PURE__*/__webpack_require__.n(_Users_pandihai_code_webpackPlugins_injectCss_webpack_plugin_tests_support_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE__);\\\\n// @ts-nocheck\\\\n\\\\n\\\\n\\\\nconsole.log(_element__WEBPACK_IMPORTED_MODULE_0__[/* Button */ \\\\\\"a\\\\\\"])   // only import button.css\\\\n\\\\n\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/normalImportWithTreeeShaking.js?\\");

/***/ })

/******/ });"
`;

exports[`test test normal import file: normalImportFile.js 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./__tests__/support/normalImportFile.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./__tests__/support/element/button.js\\":
/*!*********************************************!*\\\\
  !*** ./__tests__/support/element/button.js ***!
  \\\\*********************************************/
/*! exports provided: default */
/*! exports used: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"\\\\nlet Button = 'Button'\\\\n/* harmony default export */ __webpack_exports__[\\\\\\"a\\\\\\"] = (Button);\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/button.js?\\");

/***/ }),

/***/ \\"./__tests__/support/element/index.js\\":
/*!********************************************!*\\\\
  !*** ./__tests__/support/element/index.js ***!
  \\\\********************************************/
/*! exports provided: Message, Button */
/*! exports used: Button, Message */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"/* harmony import */ var _button__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./button */ \\\\\\"./__tests__/support/element/button.js\\\\\\");\\\\n/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\\\\\"a\\\\\\", function() { return _button__WEBPACK_IMPORTED_MODULE_0__[\\\\\\"a\\\\\\"]; });\\\\n\\\\n/* harmony import */ var _message__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ./message */ \\\\\\"./__tests__/support/element/message.js\\\\\\");\\\\n/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\\\\\"b\\\\\\", function() { return _message__WEBPACK_IMPORTED_MODULE_1__[\\\\\\"a\\\\\\"]; });\\\\n\\\\n\\\\n\\\\n\\\\n\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/index.js?\\");

/***/ }),

/***/ \\"./__tests__/support/element/message.js\\":
/*!**********************************************!*\\\\
  !*** ./__tests__/support/element/message.js ***!
  \\\\**********************************************/
/*! exports provided: default */
/*! exports used: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"\\\\nlet Message = 'Message'\\\\n/* harmony default export */ __webpack_exports__[\\\\\\"a\\\\\\"] = (Message);\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/message.js?\\");

/***/ }),

/***/ \\"./__tests__/support/element/style/button.css\\":
/*!****************************************************!*\\\\
  !*** ./__tests__/support/element/style/button.css ***!
  \\\\****************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

eval(\\"// \\\\\\".button{\\\\\\\\n    color: red;\\\\\\\\n}\\\\\\"\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/style/button.css?\\");

/***/ }),

/***/ \\"./__tests__/support/element/style/message.css\\":
/*!*****************************************************!*\\\\
  !*** ./__tests__/support/element/style/message.css ***!
  \\\\*****************************************************/
/*! no static exports found */
/***/ (function(module, exports) {

eval(\\"// \\\\\\".message{\\\\\\\\n    color: red;\\\\\\\\n}\\\\\\"\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/element/style/message.css?\\");

/***/ }),

/***/ \\"./__tests__/support/normalImportFile.js\\":
/*!***********************************************!*\\\\
  !*** ./__tests__/support/normalImportFile.js ***!
  \\\\***********************************************/
/*! no exports provided */
/*! all exports used */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
eval(\\"__webpack_require__.r(__webpack_exports__);\\\\n/* harmony import */ var _element__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./element */ \\\\\\"./__tests__/support/element/index.js\\\\\\");\\\\n/* harmony import */ var _element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE__ = __webpack_require__(/*! ./element/style/button.css */ \\\\\\"./__tests__/support/element/style/button.css\\\\\\");\\\\n/* harmony import */ var _element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE___default = /*#__PURE__*/__webpack_require__.n(_element_style_button_css__CSS__WEBPACK_IMPORTED_MODULE__);\\\\n/* harmony import */ var _element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE__ = __webpack_require__(/*! ./element/style/message.css */ \\\\\\"./__tests__/support/element/style/message.css\\\\\\");\\\\n/* harmony import */ var _element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE___default = /*#__PURE__*/__webpack_require__.n(_element_style_message_css__CSS__WEBPACK_IMPORTED_MODULE__);\\\\n// @ts-nocheck\\\\n\\\\n\\\\n\\\\nconsole.log(_element__WEBPACK_IMPORTED_MODULE_0__[/* Message */ \\\\\\"b\\\\\\"])\\\\nconsole.log(_element__WEBPACK_IMPORTED_MODULE_0__[/* Button */ \\\\\\"a\\\\\\"])\\\\n\\\\n\\\\n\\\\n//# sourceURL=webpack:///./__tests__/support/normalImportFile.js?\\");

/***/ })

/******/ });"
`;
